### PR TITLE
PERF: Series.corr/Series.cov for EA dtypes

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

```
import pandas as pd
import numpy as np

N = 1_000_000

ser1 = pd.Series(np.random.randn(N), dtype="Float64")
ser2 = pd.Series(np.random.randn(N), dtype="Float64")

%timeit ser1.corr(ser2)

# 849 ms ± 56.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)    -> main
# 33.7 ms ± 320 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  -> PR

%timeit ser1.cov(ser2)

# 739 ms ± 15.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)    -> main
# 32.9 ms ± 713 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  -> PR
```

Also, aligns the error raised for non-numeric data with that of `DataFrame.corr` and `DataFrame.cov`.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -2701,8 +2701,8 @@ def corr(
         if len(this) == 0:
             return np.nan
 
-        this_values = np.asarray(this._values)
-        other_values = np.asarray(other._values)
+        this_values = this.to_numpy(dtype=float, na_value=np.nan, copy=False)
```
Comment (jbrockmendel) [inline pandas/core/series.py:2704]: any chance this gets weird if we have non-numeric?

is this the right thing to do if we have NAs?

Comment (lukemanley) [reply pandas/core/series.py:2704]: > any chance this gets weird if we have non-numeric?

This raises for non-numeric that cannot be cast to float (same as behavior as `main` branch):

```
ser = pd.Series(["a", "b"])
ser.corr(ser)

# ValueError: could not convert string to float: 'a'
```

It allows non-numeric that can be cast to float (same behavior as `main` branch):

```
ser = pd.Series(["1", "2"])
ser.corr(ser)

# 0.9999999999999999
```
 
> is this the right thing to do if we have NAs?

NAs get dropped either way in the underlying nanops.py functions `nancorr` and `nancov`. 

FYI, this is the same approach used in `DataFrame.corr`:

https://github.com/pandas-dev/pandas/blob/0e390168407342e69886b6db7a49a281b75c9c43/pandas/core/frame.py#L10381

Comment (jbrockmendel) [reply pandas/core/series.py:2704]: > FYI, this is the same approach used in DataFrame.corr:

good point.  works for me

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**